plugin.cpp

```diff
@@ -225,7 +225,7 @@ void plugin_reconfigure(PLUGIN_HANDLE *handle, const string& newConfig)
 			}
 			else if (itr->value.IsNumber())
 			{
-				DatapointValue dpv((long) itr->value.GetInt());
+				DatapointValue dpv((long) itr->value.GetInt64());
```

Comment:
According to the C++ standard, `long` is guaranteed to be at least 32 bits. 
Therefore, using a `long` type can lead to truncation when casting data returned by `GetInt64()`, which provides 64-bit values. 
To prevent potential data loss, should we implement range checks using `IsInt()` and `IsInt64()`?

Reply:
On 64 bit architecture with UNIX like operating system long is 64 bit. On Windows it can surely be of 32 bit but that is not the case here